File: VoidMail/Models/Account.swift
import Foundation
import SwiftUI

// MARK: - User Account

struct UserAccount: Identifiable, Hashable {
    let id: String
    let email: String
    let displayName: String
    let photoURL: URL?
    let provider: AccountProvider
    var isPrimary: Bool
    var label: String
    var colorTag: AccountColor

    var initials: String {
        let parts = displayName.split(separator: " ")
        let first = parts.first?.prefix(1) ?? ""
        let last = parts.count > 1 ? parts.last!.prefix(1) : ""
        return "\(first)\(last)".uppercased()
    }

    init(id: String, email: String, displayName: String, photoURL: URL?, provider: AccountProvider, isPrimary: Bool, label: String, colorTag: AccountColor = .skyBlue) {
        self.id = id
        self.email = email
        self.displayName = displayName
        self.photoURL = photoURL
        self.provider = provider
        self.isPrimary = isPrimary
        self.label = label
        self.colorTag = colorTag
    }
}

// MARK: - Account Color Tag
// Users assign a color to each connected email account

enum AccountColor: String, CaseIterable, Identifiable, Codable, Hashable {
    case pink = "Pink"
    case green = "Green"
    case skyBlue = "Sky Blue"
    case yellow = "Yellow"
    case white = "White"

    var id: String { rawValue }

    var color: SwiftUI.Color {
        switch self {
        case .pink: return .accentPink
        case .green: return .accentGreen
        case .skyBlue: return .accentSkyBlue
        case .yellow: return .accentYellow
        case .white: return .textPrimary
        }
    }

    var icon: String {
        "circle.fill"
    }
}

// MARK: - Account Provider

enum AccountProvider: String, Hashable {
    case google = "Google"
    case imap = "IMAP"

    var icon: String {
        switch self {
        case .google: return "g.circle.fill"
        case .imap: return "envelope.circle.fill"
        }
    }
}

// MARK: - AI Alert

struct AIAlert: Identifiable {
    let id: String
    let type: AlertType
    let title: String
    let subtitle: String
    let emailId: String?
    let eventId: String?
    let date: Date

    enum AlertType {
        case awaitingReply
        case upcomingMeeting
        case followUp
        case newSender

        var icon: String {
            switch self {
            case .awaitingReply: return "exclamationmark.bubble.fill"
            case .upcomingMeeting: return "calendar.badge.clock"
            case .followUp: return "checkmark.circle.fill"
            case .newSender: return "person.badge.plus"
            }
        }

        var color: SwiftUI.Color {
            switch self {
            case .awaitingReply: return .accentPink
            case .upcomingMeeting: return .accentSkyBlue
            case .followUp: return .accentYellow
            case .newSender: return .accentYellow
            }
        }
    }
}

File: lib/models/account.dart
import 'package:flutter/material.dart';
import '../design_system/colors.dart';

/// Account Color Tags
enum AccountColor {
  pink('Pink', VoidColors.accentPink, Icons.circle),
  green('Green', VoidColors.accentGreen, Icons.circle),
  skyBlue('Sky Blue', VoidColors.accentSkyBlue, Icons.circle),
  yellow('Yellow', VoidColors.accentYellow, Icons.circle),
  white('White', VoidColors.textPrimary, Icons.circle);

  final String label;
  final Color color;
  final IconData icon;

  const AccountColor(this.label, this.color, this.icon);
}

/// Account Provider
enum AccountProvider {
  google('Google', Icons.g_mobiledata),
  imap('IMAP', Icons.email);

  final String label;
  final IconData icon;

  const AccountProvider(this.label, this.icon);
}

/// User Account
class UserAccount {
  final String id;
  final String email;
  final String displayName;
  final String? photoURL;
  final AccountProvider provider;
  final bool isPrimary;
  String label;
  AccountColor colorTag;
  String? accessToken;
  String? refreshToken;

  UserAccount({
    required this.id,
    required this.email,
    required this.displayName,
    this.photoURL,
    this.provider = AccountProvider.google,
    this.isPrimary = false,
    String? label,
    this.colorTag = AccountColor.pink,
    this.accessToken,
    this.refreshToken,
  }) : label = label ?? displayName;

  factory UserAccount.fromJson(Map<String, dynamic> json) {
    return UserAccount(
      id: json['id'] ?? '',
      email: json['email'] ?? '',
      displayName: json['displayName'] ?? json['name'] ?? '',
      photoURL: json['photoURL'] ?? json['picture'],
      provider: AccountProvider.google,
      isPrimary: json['isPrimary'] ?? false,
      label: json['label'],
      accessToken: json['accessToken'],
      refreshToken: json['refreshToken'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'email': email,
        'displayName': displayName,
        'photoURL': photoURL,
        'provider': provider.name,
        'isPrimary': isPrimary,
        'label': label,
        'colorTag': colorTag.name,
        'accessToken': accessToken,
        'refreshToken': refreshToken,
      };
}

/// AI Alert Types
enum AlertType {
  awaitingReply('Awaiting Reply', Icons.reply, VoidColors.accentPink),
  upcomingMeeting('Upcoming Meeting', Icons.event, VoidColors.accentSkyBlue),
  followUp('Follow Up', Icons.schedule, VoidColors.accentYellow),
  newSender('New Sender', Icons.person_add, VoidColors.accentYellow);

  final String label;
  final IconData icon;
  final Color color;

  const AlertType(this.label, this.icon, this.color);
}

/// AI Alert
class AIAlert {
  final String id;
  final AlertType type;
  final String title;
  final String subtitle;
  final String? emailId;
  final String? eventId;
  final DateTime date;

  AIAlert({
    required this.id,
    required this.type,
    required this.title,
    required this.subtitle,
    this.emailId,
    this.eventId,
    DateTime? date,
  }) : date = date ?? DateTime.now();

  IconData get icon => type.icon;
  Color get color => type.color;
}
